gateway/api/repositories/users.py

```diff
@@ -2,16 +2,42 @@
 Repository implementation for Groups model
 """
 
+import logging
 from typing import List
-from django.contrib.auth.models import Group, Permission
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AbstractUser, Group, Permission
 from django.db.models import Q
 
+from api.models import VIEW_PROGRAM_PERMISSION
 
-class UserRepository:  # pylint: disable=too-few-public-methods
+
+User = get_user_model()
+logger = logging.getLogger("gateway.repositories.user")
+
+
+class UserRepository:
     """
     The main objective of this class is to manage the access to the model
     """
 
+    def get_or_create_by_id(self, user_id: str) -> type[AbstractUser]:
```

Comment:
I also found a way to type that we are returning a user with `type[AbstractUser]`. This way the linter doesn't complain 😂